💡New command  :  tenant report settings set - closes #6247 (#6548)
pnp/cli-microsoft365: docs/docs/cmd/tenant/report/report-settings-set.mdx

import Global from '/docs/cmd/_global.mdx';

# tenant report settings set

Update tenant-level settings for Microsoft 365 reports

## Usage

```sh
m365 tenant report settings set [options]
```

## Options

```md definition-list
`-d, --displayConcealedNames <displayConcealedNames>`
: Determines whether all reports conceal user information such as usernames, groups, and sites. If false, all reports show identifiable information.

```

<Global />

## Examples

Set to display concealed user information in Microsoft 365 reports

```sh
m365 tenant report settings set --displayConcealedNames true
```

## Response

The command won't return a response on success

> 

Reviewer: I think there's two line breaks here, let's bring that down to one. (one customary line break at the end of an md-file)

import Global from '/docs/cmd/_global.mdx';

> import Tabs from '@theme/Tabs';

Reviewer: Let's remove imports that we don't use...

import Global from '/docs/cmd/_global.mdx';
import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

> # report settings set

Reviewer: ```suggestion
# tenant report settings set
```

> Update (adminReportSettings) tenant-level settings for Microsoft 365 reports

Reviewer: Let's use the description from the issue specs.

```suggestion
Update tenant-level settings for Microsoft 365 reports
```

## Usage

```sh
m365 tenant report settings set [options]
```

## Options

```md definition-list

> `-h, --hideUserInformation  <hideUserInformation>`

Reviewer: ```suggestion
`-h, --hideUserInformation <hideUserInformation>`
```

> : Determines whether all reports conceal user information such as usernames, groups, and sites.  

Reviewer: ```suggestion
: Determines whether all reports conceal user information such as usernames, groups, and sites. If false, all reports show identifiable information.
```

> `--verbose`

Reviewer: You can remove --verbose and --debug. They are part of the `<Global />` component

: Runs command with verbose logging

`--debug`
: Runs command with debug logging

```

<Global />

## Remarks

> Please refer to the [Graph documentation page](https://learn.microsoft.com/en-in/graph/api/adminreportsettings-update?view=graph-rest-1.0&tabs=http).

Reviewer: Let's remove this link. It's not necessary to refer API documentation that we use. We only add links that really add value for the end-user, like extra explanation about reports or something... It's not strictly necessary here I think though.

:::note

> This command returns a `204 No content` response code. To see the default logs, we recommed you to run the command with `---verbose or --debug` options

Reviewer: Let's remove this note as well. If the command returns no content, that is added to the Response section. And information about --verbose is documented elsewhere.

## Examples

> Set `displayConcealedNames` to true. 

Reviewer: Let's use the description from the issue specs.

```suggestion
Conceal user information in Microsoft 365 reports 
```

```sh
m365 tenant report settings set --hideUserInformation true
```

> Set `displayConcealedNames` to false. 

Reviewer: Let's remove this second example. It doesn't add anything, we already know how to execute the command with this option. A second example would be useful if you'd have more options.

```sh
m365 tenant report settings set --hideUserInformation false
```

## Response

> :::note

Reviewer: This note line can be removed.

> This command returns a `204 No content` response code.

Reviewer: Let's use what we use on other similar commands with no output:

```suggestion
The command won't return a response on success.
```

> <Tabs>

Reviewer: We only use tabs in the docs if we actually have output, in this case the command does not have output, so we can leave this out.
